src/java.base/share/classes/sun/security/ec/XDHKeyAgreement.java

```diff
@@ -207,9 +209,9 @@ protected SecretKey engineGenerateSecret(String algorithm)
             throw new NoSuchAlgorithmException("Algorithm must not be null");
         }
 
-        if (!(algorithm.equals("TlsPremasterSecret"))) {
+        if (!KeyUtil.isSupportedKeyAgreementOutputAlgorithm(algorithm)) {
             throw new NoSuchAlgorithmException(
-                    "Only supported for algorithm TlsPremasterSecret");
+                    "Unsupported secret key algorithm " + algorithm);
```

Comment:
Might be clearer to add a `:` after algorithm, also more consistent with DHKeyAgreement.